frontend/frontend-nav-menu.php

```diff
@@ -119,26 +118,34 @@ public function wp_get_nav_menu_items( $items ) {
 					$item->title = $this->get_item_title( $this->curlang->get_display_flag(), $name, $options );
 					$item->attr_title = '';
 					$item->classes = array( 'pll-parent-menu-item' );
+					$item->menu_order += $offset;
 					$new_items[] = $item;
 					$offset++;
 				}
 
+				$i = 0; // for incrementation of menu order only in case of dropdown
 				foreach ( $the_languages as $lang ) {
+					$i++;
 					$lang_item = clone $item;
 					$lang_item->ID = $lang_item->ID . '-' . $lang['slug']; // A unique ID
 					$lang_item->title = $this->get_item_title( $lang['flag'], $lang['name'], $options );
 					$lang_item->attr_title = '';
 					$lang_item->url = $lang['url'];
 					$lang_item->lang = $lang['locale']; // Save this for use in nav_menu_link_attributes
 					$lang_item->classes = $lang['classes'];
-					$lang_item->menu_order += $offset + $i++;
+					if ( ! empty( $options['dropdown'] ) ) {
+						$lang_item->menu_order = $item->menu_order + $i;
+					} else {
+						$lang_item->menu_order += $offset;
+					}
 					if ( ! empty( $options['dropdown'] ) ) {
 						$lang_item->menu_item_parent = $item->db_id;
 						$lang_item->db_id = 0; // to avoid recursion
 					}
```

Comment:
We can make it a bit simpler ;)

```suggestion
					if ( ! empty( $options['dropdown'] ) ) {
						$lang_item->menu_item_parent = $item->db_id;
						$lang_item->menu_order = $item->menu_order + $i;
						$lang_item->db_id = 0; // to avoid recursion
					} else {
						$lang_item->menu_order += $offset;
					}
```

Reply:
Yes indeed ! Thanks @Screenfeed , I committed your suggestion :)